aws-java-sdk-s3/src/main/java/com/amazonaws/services/s3/model/GetObjectRequest.java

```diff
@@ -121,7 +123,25 @@
      * @see GetObjectRequest#GetObjectRequest(String, String, boolean)
      */
     public GetObjectRequest(String bucketName, String key) {
-        this(bucketName, key, null);
+        this(bucketName, key, (String) null);
+    }
+
+    /**
+     * Constructs a new {@link GetObjectRequest} with all the required parameters.
+     *
+     * @param bucketName
+     *            The name of the bucket containing the desired object.
+     * @param key
+     *            The key in the specified bucket under which the object is
+     *            stored.
+     * @param lastModifiedTime
+     *            Last modified time for the object known by the client
+     *
+     * @see GetObjectRequest#GetObjectRequest(String, String, String)
+     * @see GetObjectRequest#GetObjectRequest(String, String, boolean)
+     */
+    public GetObjectRequest(String bucketName, String key, Long lastModifiedTime) {
```

Comment:
Well, if you can't accept changes in GetObjectRequest is to modify doDownload in TransferManager and pass this timestamp as a separate parameter instead of part of GetObjectRequest. Will you accept such approach?